repos/system_upgrade/el7toel8/libraries/dnfplugin.py

```diff
@@ -0,0 +1,199 @@
+import contextlib
+import itertools
+import json
+import os
+import shutil
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common import guards, mounting, overlaygen, utils
+from leapp.libraries.stdlib import CalledProcessError, api, config
+from leapp.models import FilteredRpmTransactionTasks, TargetUserSpaceInfo, UsedTargetRepositories
+
+DNF_PLUGIN_NAME = 'rhel_upgrade.py'
+DNF_PLUGIN_PATH = os.path.join('/lib/python3.6/site-packages/dnf-plugins', DNF_PLUGIN_NAME)
+DNF_PLUGIN_DATA_NAME = 'dnf-plugin-data.txt'
+DNF_PLUGIN_DATA_PATH = os.path.join('/var/lib/leapp', DNF_PLUGIN_DATA_NAME)
+DNF_PLUGIN_DATA_LOG_PATH = os.path.join('/var/log/leapp', DNF_PLUGIN_DATA_NAME)
+DNF_DEBUG_DATA_PATH = '/var/log/leapp/dnf-debugdata/'
+
+
+def install(target_basedir):
+    """ Installs our plugin to the DNF plugins """
+    try:
+        shutil.copy2(
+            api.get_file_path(DNF_PLUGIN_NAME),
+            os.path.join(target_basedir, DNF_PLUGIN_PATH.lstrip('/')))
+    except EnvironmentError as e:
+        api.current_logger().debug('Failed to install DNF plugin', exc_info=True)
+        raise StopActorExecutionError(
+            message='Failed to install DNF plugin. Error: {}'.format(e.message)
+        )
+
+
+def build_plugin_data(target_repoids, debug, test):
+    """
+    Generates a dictionary with the DNF plugin data.
+    """
+    # get list of repo IDs of target repositories that should be used for upgrade
+    tasks = next(api.consume(FilteredRpmTransactionTasks), FilteredRpmTransactionTasks())
+
+    data = {
+        'pkgs_info': {
+            'local_rpms': [os.path.join('/installroot', pkg.lstrip('/')) for pkg in tasks.local_rpms],
+            'to_install': [pkg for pkg in tasks.to_install],
+            'to_remove': [pkg for pkg in tasks.to_remove],
+            #            'to_upgrade': [pkg for pkg in tasks.to_upgrade]
```

Comment:
Why this one is commented out? (nowadays, `to_upgrade` is `to_keep`)

Reply:
There is a pending PR with this and I already had it prepared, to_keep isn't used by the DNF plugin, this was to keep me reminded that this has to be updated in case 226 gets merged before.

https://github.com/oamg/leapp-repository/pull/226

Reply:
Would you put there at least FIXME/TODO msg so in case this will be merged, we will be sure nobody forget about that? Possibly this will be merged before #226